xrmocap/core/visualization/visualize_smpl.py

```diff
@@ -141,41 +142,54 @@ def visualize_smpl_data(
         camera_parameter=cam_param,
         device=device,
         logger=logger)
-    # check whether to write video directly or write images first
-    if check_path_suffix(output_path, '.mp4'):
-        write_video = True
-        if batch_size < data_len:
-            output_dir = f'{output_path}_temp'
-            os.makedirs(output_dir, exist_ok=True)
-            write_img = True
-            remove_output_dir = True
-        else:
-            write_img = False
-            remove_output_dir = False
-    else:
-        write_video = False
-        output_dir = output_path
-        write_img = True
-        remove_output_dir = False
-    mperson_verts = None
-    for person_idx in range(n_person):
-        smpl_data = smpl_data_list[person_idx]
-        param_dict = smpl_data.to_tensor_dict(device=device)
-        model = body_model_dict[smpl_data['gender']]
-        body_model_output = model(**param_dict)
-        verts = body_model_output['vertices']  # n_batch, n_verts, 3
-        sperson_verts = torch.unsqueeze(verts, dim=1)
-        # sperson_verts.shape: n_batch, n_person, n_verts, 3
-        if mperson_verts is None:
-            mperson_verts = sperson_verts
-        else:
-            mperson_verts = torch.cat((mperson_verts, sperson_verts), dim=1)
-    file_names_cache = None
-    output_img_list = []
-    for start_idx in tqdm(
-            range(0, data_len, batch_size), disable=disable_tqdm):
-        end_idx = min(start_idx + batch_size, data_len)
-        # prepare background array for this batch
+    # check whether to write video or write images
+    write_video, write_img = (True, False) \
+        if check_path_suffix(output_path, '.mp4') else (False, True)
+    if write_video:
+        xrprimer_video_writer = VideoWriter(
+            output_path, [cam_param.height, cam_param.width])
+
+    total_iter = data_len // batch_size + 1
+    curr_iter = 0
+
+    while (curr_iter < total_iter):
+
+        mperson_verts = None
+        start_idx = curr_iter * batch_size
+        end_idx = min(data_len, start_idx + batch_size)
+        for person_idx in range(n_person):
+            smpl_data = smpl_data_list[person_idx]
+            param_dict = smpl_data.to_tensor_dict(device=device)
+
+            param_dict_input = dict()
+            for key, value in param_dict.items():
+                if key not in ['right_hand_pose', 'left_hand_pose']:
+                    param_dict_input[key] = torch.tensor(
```

Comment:
`dtype` and `device` have already been converted at https://github.com/openxrlab/xrmocap/blob/main/xrmocap/data_structure/body_model/smplx_data.py#L380. This loop can be removed.